src/raftstore/store/worker/apply.rs

```diff
@@ -195,48 +194,131 @@ impl ApplyCallback {
 
 struct ApplyContext<'a> {
     host: &'a CoprocessorHost,
-    wb: WriteBatch,
+    wb: Option<WriteBatch>,
     cbs: MustConsumeVec<ApplyCallback>,
+    apply_res: Vec<ApplyRes>,
     wb_last_bytes: u64,
     wb_last_keys: u64,
+    committed_count: usize,
+    // `sync_log_hint` is a global hint indicate that whether synchronize wal
+    // when data is written to kv engine.
+    sync_log_hint: bool,
```

Comment:
Seems the only use case is that, if `sync_log && sync_log_hint`, we need to synchronize WAL, can we combine them together?

Reply:
It's not easy when using bool. `sync_log_hint` is set by configuration, and `sync_log` is set depend on the content of entries being applied.